experimental/rsocket-src/transports/TcpConnectionFactory.cpp

```diff
@@ -0,0 +1,128 @@
+// Copyright 2004-present Facebook. All Rights Reserved.
+
+#include <folly/io/async/AsyncSocket.h>
+#include <folly/io/async/EventBaseManager.h>
+#include <folly/io/async/ScopedEventBaseThread.h>
+#include <rsocket/transports/TcpConnectionFactory.h>
+#include "src/framed/FramedDuplexConnection.h"
+#include "src/tcp/TcpDuplexConnection.h"
+
+using namespace ::reactivesocket;
+using namespace ::folly;
+
+namespace rsocket {
+
+// carry the thread/eventbase with the DuplexConnection
+class FramedDuplexConnectionOnThread : public DuplexConnection {
+ public:
+  FramedDuplexConnectionOnThread(
+      std::unique_ptr<FramedDuplexConnection> fd,
+      std::unique_ptr<ScopedEventBaseThread> eventBaseThread)
+      : fd_(std::move(fd)), eventBaseThread_(std::move(eventBaseThread)) {}
+
+  ~FramedDuplexConnectionOnThread() {
+    LOG(INFO) << "FramedDuplexConnectionOnThread => destroy";
+  }
+
+  void setInput(
+      std::shared_ptr<Subscriber<std::unique_ptr<folly::IOBuf>>> framesSink) {
+    fd_->setInput(std::move(framesSink));
+  }
+
+  std::shared_ptr<Subscriber<std::unique_ptr<folly::IOBuf>>> getOutput() {
+    return fd_->getOutput();
+  }
+
+ private:
+  std::unique_ptr<FramedDuplexConnection> fd_;
+  std::unique_ptr<ScopedEventBaseThread> eventBaseThread_;
```

Comment:
My goal has been for the transport layer to control it's concurrency. Thus, the TcpConnectionFactory chooses threading. 

Right now the impl is very naive and creates a new thread for every connection. I want to pass in a config for number of threads, and default to a single thread. 

Then, each connection would be created on one of the threads in the pool ... or on just the single thread.